modules/apps/27-interchain-accounts/keeper/handshake.go

```diff
@@ -71,10 +75,12 @@ func (k Keeper) OnChanOpenTry(
 	if order != channeltypes.ORDERED {
 		return sdkerrors.Wrapf(channeltypes.ErrInvalidChannelOrdering, "invalid channel ordering: %s, expected %s", order.String(), channeltypes.ORDERED.String())
 	}
-	if version != types.Version {
-		return sdkerrors.Wrapf(types.ErrInvalidVersion, "got: %s, expected %s", version, types.Version)
+
+	if !strings.Contains(version, types.Version) {
+		return sdkerrors.Wrapf(types.ErrInvalidVersion, "invalid version: %s, expected %s", version, types.Version)
 	}
-	if counterpartyVersion != types.Version {
+
+	if !strings.Contains(counterpartyVersion, types.Version) {
```

Comment:
With the inclusion of the account address in the version string we now have to deviate from the spec as we cannot explicitly check for equality.

I've updated these checks to use `strings.Contains()`, but perhaps there's some room for discussion on this topic

Reply:
`Contains` may be okay. I'm wondering though should we be more strict about this and have some kind of validate version string helper function to ensure the format we expect is being passed in? 

@colin-axner would like to hear your thoughts

Reply:
I think this would be a nicer option. Just basic assertion of `${types.Version}-${validateBech32addr}`

Reply:
Yes I'd definitely like stronger validation. The two areas we do validation now is in [24-host](https://github.com/cosmos/ibc-go/blob/main/modules/core/24-host/validate.go) and [identifier parsing](https://github.com/cosmos/ibc-go/blob/main/modules/core/02-client/types/keys.go#L45)

I think the address validation should be light weight. I'm not sure it needs to be a valid bech 32 address since this can be address on a different chain. Maybe just a string which doesn't contain a `-`? It'd be nice if the interchain accounts mentioned this validation

I'm also okay doing some arbitrary address length validation and increasing the maximum allowed length if someone runs into an issue 

Adding a validation function also increases the readability of the code

Reply:
Updated! [See here](https://github.com/cosmos/ibc-go/pull/410#discussion_r710132939)